### DISCUSSION TITLE
Add download file link to a column view

### DISCUSSION THREAD

**Discussion by mouseman85** [microsoft-search/pnp-modern-search#1758] [Q&A] [answered]
Is it possible to add a download file link to a column view? My users have a big library of PDFs that they search and will always need to just download the file when they find it. It would save them time if they can just click a link in a column view that allows them to download that file. I am new to handlebar so I'm not sure what sort of code would be needed for this.

**Comment by wobba** [comment]
We currently don't have this capability. You can probably modify the template to point to a valid download link if you find the format of what that link should look like. Could work by doing:

`<a href="{{slot item @root.slots.DefaultEncodingURL}}" download>download</a>`

**Comment by mouseman85** [comment, marked as answer]
Just to help anyone else that might need this I worked out the file path to create a Download link in a column view below is what I'm using as the handlebar code this opens a blank tab and downloads the file.

`<a href="https://tenant.sharepoint.com/sites/SITENAME/DOCUMENTLIBRARYNAME/_layouts/15/download.aspx?sourceurl={{slot item @root.slots.Path}}" data-interception="off" target="_blank">Download</a>`

**Reply by wobba** [reply]
.Path will likely fail for pdf's and why I suggested to use DefaultEncodingURL.

**Reply by mouseman85** [reply]
Tested with PDFs and worked fine. The DefaultEncodingURL didn’t work it would download but the file would be a .htm page and always called main.htm.

I’m not sure if this is because of how I’ve had to build the download url to get it to work. But I couldn’t get a correct result with DefaultEncodingURL

**Reply by wobba** [reply]
All good then 🙂 Could be the slot mapping code which handles this - didn't actually check this 😉
